src/Equations/elastic/Kernels/DeviceAux/cuda/KernelsAux.cu

```diff
@@ -0,0 +1,335 @@
+#include <Kernels/precision.hpp>
+#include <init.h>
+#include <tensor.h>
+#include <yateto.h>
+#include <cuda.h>
+#include <cstdio>
+
+
+namespace seissol::kernels::local_flux::aux::details {
+
+__global__ void kernelFreeSurfaceGravity(
+  real** dofsFaceBoundaryNodalPtrs,
+  real** displacementDataPtrs,
+  double* rhos,
+  double g,
+  size_t numElements) {
+
+  const int tid = threadIdx.x;
+  const int elementId = blockIdx.x;
+  if (elementId < numElements) {
+    const double rho = rhos[elementId];
+    real* elementBoundaryDofs = dofsFaceBoundaryNodalPtrs[elementId];
+    real* elementDisplacement = displacementDataPtrs[elementId];
+
+    constexpr auto numNodes = seissol::nodal::tensor::nodes2D::Shape[0];
+    if (tid < numNodes) {
+      constexpr auto INodalStride = yateto::leadDim<seissol::init::INodal>();
+
+      const auto pressureAtBnd = real(-1.0) * rho * g * elementDisplacement[tid];
+
+      #pragma unroll
+      for (int component{0}; component < 3; ++component) {
+        elementBoundaryDofs[tid + component * INodalStride] =
+          2.0 * pressureAtBnd - elementBoundaryDofs[tid + component * INodalStride];
+      }
+    }
+  }
+}
+
+void launchFreeSurfaceGravity(real** dofsFaceBoundaryNodalPtrs,
+                              real** displacementDataPtrs,
+                              double* rhos,
+                              double g,
+                              size_t numElements,
+                              void* deviceStream) {
+  dim3 block(yateto::leadDim<seissol::nodal::init::nodes2D>(), 1, 1);
+  dim3 grid(numElements, 1, 1);
+  auto stream = reinterpret_cast<cudaStream_t>(deviceStream);
+  kernelFreeSurfaceGravity<<<grid, block, 0, stream>>>(dofsFaceBoundaryNodalPtrs,
+                                                       displacementDataPtrs,
+                                                       rhos,
+                                                       g,
+                                                       numElements);
+}
+
+
+__global__ void  kernelEasiBoundary(real** dofsFaceBoundaryNodalPtrs,
+                                    real** easiBoundaryMapPtrs,
+                                    real** easiBoundaryConstantPtrs,
+                                    size_t numElements) {
+
+  const int tid = threadIdx.x;
+  const int elementId = blockIdx.x;
+
+  constexpr auto leadINodalDim = yateto::leadDim<seissol::init::INodal>();
+  constexpr auto INodalDim0 = seissol::tensor::INodal::Shape[0];
+  constexpr auto INodalDim1 = seissol::tensor::INodal::Shape[1];
+  __shared__ __align__(8) real resultTerm[INodalDim1][INodalDim0];
```

Comment:
does cuda support alignas(8)?

Reply:
I don't know. In general, it is a piece of nonsense to align the shared memory. The compiler was supposed to do it. All in all,  `__align__(8)` was super crucial for A100 performance. I just don't want to even touch it.  Somehow `__align__(8)` significantly influences the instruction scheduling of `nvcc` compiler.

Reply:
Ah, I know this type of code...